prow/jenkins/jenkins.go

```diff
@@ -292,3 +297,29 @@ func (c *Client) GetLog(job string, build int) ([]byte, error) {
 	}
 	return buf, nil
 }
+
+// Abort a Jenkins build that is either running or waiting in the queue.
+func (c *Client) Abort(job string, build *JenkinsBuild) error {
+	c.log("Abort", job, build)
+	u := fmt.Sprintf("%s/job/%s/%d/stop", c.baseURL, job, build.Number)
+	var header map[string]string
+	if build.IsEnqueued() {
+		// Openshift CI runs a Jenkins proxy that uses a local cache of job names
+		// to proxy requests to multiple Jenkins masters. In case there is a queued
+		// build that needs to be cancelled, the proxy has no idea what is the
+		// correct master to redirect to because the queue cancellation request bears
+		// no information about the job for which the build needs to be cancelled.
+		// The custom header is used by the proxy to figure the correct master out.
+		header = map[string]string{"Jenkins-Job": job}
```

Comment:
@stevekuznetsov I don't like this at all. How about using a `max_concurrency` in the operator that will enables us to stay away from filling the queue? Cancelling superseded running jobs works like a charm.

Reply:
Not sure I follow the comment -- isn't the `POST` containing the job name and build number?

Reply:
Discussed in IRC, we can filter these prowjobs out and not terminate any duplicates as long as they are in the queue.